setup-local/src/binaryControl.js

```diff
@@ -0,0 +1,232 @@
+   */
+  _generateLogFileMetadata() {
+    this.logFileName = process.env[BROWSERSTACK_LOCAL_LOGS_FILE] || `${LOCAL_LOG_FILE_PREFIX}_${github.context.job}_${Date.now()}.log`;
+    this.logFilePath = path.resolve(this.binaryFolder, this.logFileName);
+    core.exportVariable(BROWSERSTACK_LOCAL_LOGS_FILE, this.logFileName);
+  }
+
+  /**
+   * Generates the args to be provided for the Local Binary based on the operation, i.e.
+   * start/stop.
+   * These are generated based on the input state provided to the Binary Control.
+   */
+  _generateArgsForBinary() {
+    const {
+      accessKey: key,
+      localArgs,
+      localIdentifier,
+      localLoggingLevel: verbose,
+      localTesting: binaryAction,
+    } = this.stateForBinary;
+
+    let argsString = `--key ${key} --only-automate --ci-plugin GitHubAction `;
+
+    switch (binaryAction) {
+      case LOCAL_TESTING.START: {
+        if (localArgs) argsString += `${localArgs} `;
+        if (localIdentifier) argsString += `--local-identifier ${localIdentifier} `;
+        if (verbose) {
+          this._generateLogFileMetadata();
+          argsString += `--verbose ${verbose} --log-file ${this.logFilePath} `;
+        }
+        break;
+      }
+      case LOCAL_TESTING.STOP: {
+        if (localIdentifier) argsString += `--local-identifier ${localIdentifier} `;
+        break;
+      }
+      default: {
+        throw Error('Invalid Binary Action');
+      }
+    }
+
+    this.binaryArgs = argsString;
+  }
+
+  /**
+   * Triggers the Local Binary. It is used for starting/stopping.
+   * @param {String} operation start/stop operation
+   */
+  async _triggerBinary(operation) {
+    let triggerOutput = '';
+    let triggerError = '';
+    await exec.exec(
+      `${LOCAL_BINARY_NAME} ${this.binaryArgs} --daemon ${operation}`,
+      [],
+      {
+        listeners: {
+          stdout: (data) => {
+            triggerOutput += data.toString();
+          },
+          stderr: (data) => {
+            triggerError += data.toString();
+          },
+        },
+      },
+    );
+
+    return {
+      output: triggerOutput,
+      error: triggerError,
+    };
+  }
+
+  /**
+   * Downloads the Local Binary, extracts it and adds it in the PATH variable
+   */
+  async downloadBinary() {
+    try {
+      if (Utils.checkToolInCache(LOCAL_BINARY_NAME)) {
+        core.info('BrowserStackLocal binary already exists in cache. Using that instead of downloading again...');
+        return;
+      }
+      await this._makeDirectory();
+      core.info('Downloading BrowserStackLocal binary...');
+      const downloadPath = await tc.downloadTool(this.binaryLink, path.resolve(this.binaryFolder, 'binaryZip'));
+      const extractedPath = await tc.extractZip(downloadPath, this.binaryFolder);
+      core.info(`BrowserStackLocal binary downloaded & extracted successfuly at: ${extractedPath}`);
+      const cachedPath = await tc.cacheDir(extractedPath, LOCAL_BINARY_NAME, '1.0.0');
```

Comment:
does `1.0.0` denotes cached version?

Reply:
yes. it doesn't play much of a role here. It required the `version` input, thus gave it as `1.0.0`. We aren't caching multiple versions etc.

Reply:
resolving this.